src/main/java/com/scalar/cassy/transferer/AzureBlobFileDownloader.java

```diff
@@ -52,10 +53,18 @@ public void download(RestoreConfig config) {
                     throw new FileTransferException(
                         "Download file failed : " + destFile.toString(), error);
                   }));
+
```

Comment:
https://github.com/scalar-labs/cassy/pull/75/files#diff-1bb8899e39b39ed23e3e64d77360ec8cf65a5b59d1a8e314288ce5b7648f4047R38

I think it smells a bit since it uses `for` for `Iterable`. Does it expand the whole files?
Can you try using the iterator simply instead of `for`?

If it doesn't work, I have no clue.
Let's try to use `BlobContainerClient` instead of `BlobContainerAsyncClient` and downloading file synchronously one by one.

Reply:
@feeblefakie san,
Blob file download implementation modified with `BlobContainerAsyncClient`.
Restoration is working fine with 64GB of data.
Could you please check when you get a chance?